libglusterfs/src/syscall.c

```diff
@@ -362,7 +364,9 @@ sys_writev(int fd, const struct iovec *iov, int iovcnt)
 ssize_t
 sys_read(int fd, void *buf, size_t count)
```

Comment:
Who says we only read strings here? What if it was binary data?

Reply:
Sorry. Actually the coverity issue mentions it as "String not null terminated". I guess the null character should be added in the function calling sys_read which passes a char buffer as argument. And i guess in the case of sys_readlink null character can be appended right?

Reply:
Correct. 'man 2 readlink' indeed says:
```
readlink() does not append  a  null byte  to buf.  It will (silently) truncate the contents (to a length of
bufsiz characters), in case the buffer is too small to hold all of  the contents.
```